homeassistant/components/lacrosse_view/config_flow.py

```diff
@@ -0,0 +1,123 @@
+"""Config flow for LaCrosse View integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from lacrosse_view import LaCrosse, Location, LoginError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required("username"): str,
+        vol.Required("password"): str,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> list[Location]:
+    """Validate the user input allows us to connect."""
+
+    api = LaCrosse(async_get_clientsession(hass))
+
+    try:
+        if not await api.login(data["username"], data["password"]):
+            raise InvalidAuth
+
+        locations: list[Location] = await api.get_locations()
+        if not locations:
+            raise NoLocations(
+                "No locations found for account {}".format(data["username"])
+            )
+    except LoginError as error:
+        raise InvalidAuth from error
+
+    return locations
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for LaCrosse View."""
+
+    VERSION = 1
+    data: dict[str, str] = {}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        try:
+            info = await validate_input(self.hass, user_input)
+        except InvalidAuth:
+            errors["base"] = "invalid_auth"
+        except NoLocations:
+            errors["base"] = "no_locations"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+        else:
+            self.data = user_input
+            return self.async_show_form(
+                step_id="location",
+                data_schema=vol.Schema(
+                    {
+                        vol.Required("location"): vol.In(
+                            {
+                                f"{location.id} {location.name}": location.name
```

Comment:
I'd maybe say instead to just have the id here as the key? Rather than encoding information in the key, store the locations and get the name back from the id.